app/src/main/java/org/thoughtcrime/securesms/onboarding/messagenotifications/MessageNotificationsActivity.kt

```diff
@@ -0,0 +1,172 @@
+import androidx.annotation.StringRes
+import androidx.compose.foundation.BorderStroke
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.ButtonDefaults
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.OutlinedButton
+import androidx.compose.material.RadioButton
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.ComposeView
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.tooling.preview.PreviewParameter
+import androidx.compose.ui.unit.dp
+import dagger.hilt.android.AndroidEntryPoint
+import network.loki.messenger.R
+import org.session.libsession.utilities.TextSecurePreferences
+import org.thoughtcrime.securesms.ApplicationContext
+import org.thoughtcrime.securesms.BaseActionBarActivity
+import org.thoughtcrime.securesms.home.HomeActivity
+import org.thoughtcrime.securesms.notifications.PushRegistry
+import org.thoughtcrime.securesms.ui.AppTheme
+import org.thoughtcrime.securesms.ui.OutlineButton
+import org.thoughtcrime.securesms.ui.PreviewTheme
+import org.thoughtcrime.securesms.ui.ThemeResPreviewParameterProvider
+import org.thoughtcrime.securesms.ui.contentDescription
+import org.thoughtcrime.securesms.ui.h8
+import org.thoughtcrime.securesms.ui.h9
+import org.thoughtcrime.securesms.ui.session_accent
+import org.thoughtcrime.securesms.ui.small
+import org.thoughtcrime.securesms.util.setUpActionBarSessionLogo
+import javax.inject.Inject
+
+@AndroidEntryPoint
+class MessageNotificationsActivity : BaseActionBarActivity() {
+
+    @Inject lateinit var pushRegistry: PushRegistry
+
+    private val viewModel: MessageNotificationsViewModel by viewModels()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setUpActionBarSessionLogo(true)
+        TextSecurePreferences.setHasSeenWelcomeScreen(this, true)
+
+        ComposeView(this)
+            .apply { setContent { MessageNotificationsScreen() } }
+            .let(::setContentView)
+    }
+
+    @Composable
+    private fun MessageNotificationsScreen() {
+        val state by viewModel.stateFlow.collectAsState()
+
+        AppTheme {
+            MessageNotificationsScreen(state, viewModel::setEnabled, ::register)
+        }
+    }
+
+    private fun register() {
```

Comment:
Could we make sure we add comments, even when it feels obvious, just to help understand what is being done and why?

Reply:
part of onboarding